mantidproject/mantid
# Adding muon data plot pane (making muon plotting easier)

This is the first set of changes for the muon plotting window. This series of work is to decouple the core part of the plotting from the current muon GUI, making it more flexible. I am also attempting to clean up the code to make it easier to follow. 



**To test:**
1. Open the build and make sure the muon analysis and frequency domain analysis are the same as release (these changes should not be visible)

2. Replace the muon analysis file (ask me to send it to you)
3. Open muon analysis
4. Make sure that plotting data works (fits will not work).:
5. Load current run
6. Stepping to the next/previous run
7. Check rebin and the plot raw checkbox
8. Check the counts/asymmetry selection works
9. Switching the deadtime table (on or off), its best to look at the counts for this
10. Check adding/removing groups works
11. Check adding a phasequad works
12. Check tiled plots work (including the tiled by option)



Fixes #31494. 



This does not need to be in the release notes as it does not alter the user experience. 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
